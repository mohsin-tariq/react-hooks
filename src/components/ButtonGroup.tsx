import styled from 'styled-components'

const ButtonGroup = styled.div`
  display: flex;
  button:not(:last-child) {
    margin-right: 1rem;
  }
`

export default ButtonGroup
