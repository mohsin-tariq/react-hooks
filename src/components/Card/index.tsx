export { default as Card } from './Card'
export { default as RequestTitle } from './RequestTitle'
export { default as RequestDetail } from './RequestDetail'
export { default as RequestControls } from './RequestControls'
