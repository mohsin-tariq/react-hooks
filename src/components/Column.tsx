import styled from 'styled-components'
import { breakpoints } from './tools'

const Column = styled.div`
  margin: 0 2rem;

  ${breakpoints.sm} {
    width: 100%;
  }
`

export default Column
