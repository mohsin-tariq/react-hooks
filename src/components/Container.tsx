import styled from 'styled-components'

const Container = styled.div`
  display: flex;
  width: 100%;
  max-width: 120rem;
  margin: 2rem;
  justify-content: center;
`

export default Container
