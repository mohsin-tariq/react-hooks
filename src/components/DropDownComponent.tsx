import { memo } from 'react'
import { Zones } from '../features/timezone/types'
import { selectors } from '../features/timezone'

interface Props {
  zoneList: Zones[]
  setZoneDetail: React.Dispatch<React.SetStateAction<Zones>>
}

const DropDownComponent = ({ zoneList, setZoneDetail }: Props): JSX.Element => {
  const handleOnChange = (e) => {
    setZoneDetail(
      zoneList.filter(
        (item) => selectors.getSelector(item) === e.target.value
      )[0]
    )
  }
  return (
    <select onChange={(e) => handleOnChange(e)}>
      {zoneList.map((item) => (
        <option key={item.zoneName}>{selectors.getSelector(item)}</option>
      ))}
    </select>
  )
}

export default memo(DropDownComponent)
