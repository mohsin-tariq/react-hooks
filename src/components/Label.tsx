import styled from 'styled-components'

const Label = styled.label`
  color: #767676;
  font-size: 1.2rem;
  line-height: 1.5rem;
  font-weight: 400;
  display: block;
  margin-bottom: 0.8rem;
`

export default Label
