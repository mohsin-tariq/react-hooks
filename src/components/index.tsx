export * from './Buttons'
export * from './Timezone'
export * from './Card'
export { default as Container } from './Container'
export { default as Column } from './Column'
export { default as ButtonGroup } from './ButtonGroup'
export { default as Paragraph } from './Paragraph'
export { default as Title } from './Title'
export { default as Label } from './Label'
