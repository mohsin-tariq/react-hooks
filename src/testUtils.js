
/** eslint-disable **/
import { render as rtlRender } from '@testing-library/react'
import { Provider } from 'react-redux'
import configureMockStore from 'redux-mock-store'
import '@testing-library/jest-dom/extend-expect'

const initialState = {
  zone: {
    data: [
      {
        countryCode: 'AD',
        countryName: 'Andorra',
        zoneName: 'Europe/Andorra',
        gmtOffset: 3600,
        timestamp: 1614084530,
      },
      {
        countryCode: 'AL',
        countryName: 'Albania',
        zoneName: 'Europe/Albania',
        gmtOffset: 3600,
        timestamp: 1614084530,
      },
    ],
  },
}

const mockStore = configureMockStore()

function render(
  ui,
  { store = mockStore(initialState), ...renderOptions } = {}
) {
  // eslint-disable-next-line react/prop-types
  const Wrapper = ({ children }) => {
    return <Provider store={store}>{children}</Provider>
  }
  return rtlRender(ui, { wrapper: Wrapper, ...renderOptions })
}

// re-export everything
export * from '@testing-library/react'
// override render method
export { render }
