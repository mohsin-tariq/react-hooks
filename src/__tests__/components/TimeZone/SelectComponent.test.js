import { render, fireEvent, screen } from '../../../testUtils'
import { SelectComponent } from '../../../components'


describe('SelectComponent', () => {
  it('render Snapshot', () => {
    const { asFragment } = render(<SelectComponent zoneList= {[
      {
        countryCode: 'AD',
        countryName: 'Andorra',
        zoneName: 'Europe/Andorra',
        gmtOffset: 3600,
        timestamp: 1614084530,
      },
      {
        countryCode: 'AL',
        countryName: 'Albania',
        zoneName: 'Europe/Albania',
        gmtOffset: 3600,
        timestamp: 1614084530,
      }
    ]}
    setZoneDetail={jest.fn()}
    />)
    expect(asFragment()).toMatchSnapshot()
  })

  it('verify if element exists', () => {
    render(<SelectComponent  zoneList= {[
        {
          countryCode: 'AD',
          countryName: 'Andorra',
          zoneName: 'Europe/Andorra',
          gmtOffset: 3600,
          timestamp: 1614084530,
        },
        {
          countryCode: 'AL',
          countryName: 'Albania',
          zoneName: 'Europe/Albania',
          gmtOffset: 3600,
          timestamp: 1614084530,
        }
      ]}
      setZoneDetail={jest.fn()}
     />)
    expect(screen.getByText('Andorra, Europe/Andorra')).toBeInTheDocument()
  })

  it('onchange set value', () => {
    render(<SelectComponent  zoneList= {[
        {
          countryCode: 'AD',
          countryName: 'Andorra',
          zoneName: 'Europe/Andorra',
          gmtOffset: 3600,
          timestamp: 1614084530,
        },
        {
          countryCode: 'AL',
          countryName: 'Albania',
          zoneName: 'Europe/Albania',
          gmtOffset: 3600,
          timestamp: 1614084530,
        }
      ]}
      setZoneDetail={jest.fn()}
     />)
     fireEvent.change(screen.getByDisplayValue('Andorra, Europe/Andorra'), {
      target: { value: "Albania, Europe/Albania" },
    });
    expect(screen.getByDisplayValue('Albania, Europe/Albania')).toBeInTheDocument()
  })
})

