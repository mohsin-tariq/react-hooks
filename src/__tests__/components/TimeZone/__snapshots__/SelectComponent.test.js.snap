// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SelectComponent render Snapshot 1`] = `
<DocumentFragment>
  <select>
    <option>
      Andorra, Europe/Andorra
    </option>
    <option>
      Albania, Europe/Albania
    </option>
  </select>
</DocumentFragment>
`;
